pkg/ext-proc/backend/inferencepool_reconciler.go

```diff
@@ -37,26 +37,33 @@ func (c *InferencePoolReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	loggerDefault.Info("Reconciling InferencePool", "name", req.NamespacedName)
 
 	serverPool := &v1alpha1.InferencePool{}
+
 	if err := c.Get(ctx, req.NamespacedName, serverPool); err != nil {
+		if errors.IsNotFound(err) {
+			loggerDefault.Info("InferencePool not found. Clearing the datastore", "name", req.NamespacedName)
+			c.Datastore.Clear()
+			return ctrl.Result{}, nil
+		}
 		loggerDefault.Error(err, "Unable to get InferencePool", "name", req.NamespacedName)
 		return ctrl.Result{}, err
-	}
-	if c.Datastore.inferencePool == nil || !reflect.DeepEqual(serverPool.Spec.Selector, c.Datastore.inferencePool.Spec.Selector) {
-		c.updateDatastore(logger, serverPool)
-		c.Datastore.flushPodsAndRefetch(ctx, c.Client, serverPool)
-	} else {
-		c.updateDatastore(logger, serverPool)
+	} else if !serverPool.DeletionTimestamp.IsZero() {
+		loggerDefault.Info("InferencePool is marked for deletion. Clearing the datastore", "name", req.NamespacedName)
+		c.Datastore.Clear()
+		return ctrl.Result{}, nil
 	}
 
+	c.updateDatastore(ctx, serverPool)
+
 	return ctrl.Result{}, nil
 }
 
-func (c *InferencePoolReconciler) updateDatastore(logger logr.Logger, serverPool *v1alpha1.InferencePool) {
-	pool, _ := c.Datastore.getInferencePool()
-	if pool == nil ||
-		serverPool.ObjectMeta.ResourceVersion != pool.ObjectMeta.ResourceVersion {
-		logger.V(logutil.DEFAULT).Info("Updating inference pool", "target", klog.KMetadata(&serverPool.ObjectMeta))
-		c.Datastore.setInferencePool(serverPool)
+func (c *InferencePoolReconciler) updateDatastore(ctx context.Context, newPool *v1alpha1.InferencePool) {
+	logger := log.FromContext(ctx)
+	oldPool, _ := c.Datastore.PoolGet()
```

Comment:
nit: Don't ignore the error. Check the error instead of checking the nil of the oldPools

Reply:
done.